packages/puppeteer-core/src/common/Input.ts

```diff
@@ -467,84 +275,17 @@ interface MouseState {
 }
 
 /**
- * The Mouse class operates in main-frame CSS pixels
- * relative to the top-left corner of the viewport.
- * @remarks
- * Every `page` object has its own Mouse, accessible with [`page.mouse`](#pagemouse).
- *
- * @example
- *
- * ```ts
- * // Using ‘page.mouse’ to trace a 100x100 square.
- * await page.mouse.move(0, 0);
- * await page.mouse.down();
- * await page.mouse.move(0, 100);
- * await page.mouse.move(100, 100);
- * await page.mouse.move(100, 0);
- * await page.mouse.move(0, 0);
- * await page.mouse.up();
- * ```
- *
- * **Note**: The mouse events trigger synthetic `MouseEvent`s.
- * This means that it does not fully replicate the functionality of what a normal user
- * would be able to do with their mouse.
- *
- * For example, dragging and selecting text is not possible using `page.mouse`.
- * Instead, you can use the {@link https://developer.mozilla.org/en-US/docs/Web/API/DocumentOrShadowRoot/getSelection | `DocumentOrShadowRoot.getSelection()`} functionality implemented in the platform.
- *
- * @example
- * For example, if you want to select all content between nodes:
- *
- * ```ts
- * await page.evaluate(
- *   (from, to) => {
- *     const selection = from.getRootNode().getSelection();
- *     const range = document.createRange();
- *     range.setStartBefore(from);
- *     range.setEndAfter(to);
- *     selection.removeAllRanges();
- *     selection.addRange(range);
- *   },
- *   fromJSHandle,
- *   toJSHandle
- * );
- * ```
- *
- * If you then would want to copy-paste your selection, you can use the clipboard api:
- *
- * ```ts
- * // The clipboard api does not allow you to copy, unless the tab is focused.
- * await page.bringToFront();
- * await page.evaluate(() => {
- *   // Copy the selected content to the clipboard
- *   document.execCommand('copy');
- *   // Obtain the content of the clipboard as a string
- *   return navigator.clipboard.readText();
- * });
- * ```
- *
- * **Note**: If you want access to the clipboard API,
- * you have to give it permission to do so:
- *
- * ```ts
- * await browser
- *   .defaultBrowserContext()
- *   .overridePermissions('<your origin>', [
- *     'clipboard-read',
- *     'clipboard-write',
- *   ]);
- * ```
- *
- * @public
+ * @internal
  */
-export class Mouse {
+export class CDPMouse extends Mouse {
   #client: CDPSession;
-  #keyboard: Keyboard;
+  #keyboard: CDPKeyboard;
 
   /**
```

Comment:
Not needed.

Reply:
Within the implementation, we don't want to treat the type as the public type. In fact, it's preferred to use the internal type wherever you can. Reason being when the consumer accesses this class through `page.mouse` for example, they will see the signatures from the `api/Input.js` folder rather than the ones declared here.

It's useful is someone knows they have a specific implementation and wants to just `as` cast the implementation rather than for each object of the implementation.

For example, compare

```ts
(page as CDPPage).mouse as CDPMouse;
```

versus

```ts
(page as CDPPage).mouse; // already a CDPMouse
```